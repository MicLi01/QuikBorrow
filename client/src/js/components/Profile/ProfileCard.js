import React, { Component } from "react";
import styled from 'styled-components';
import { Card, CardContent, CardMedia, Typography, Box, Button, TextField } from '@material-ui/core';
import { Link } from 'react-router-dom';
import { ListItem, ListItemAvatar, Avatar, ListItemText } from '@material-ui/core';
import CheckIcon from '@material-ui/icons/Check';
import CancelIcon from '@material-ui/icons/Cancel';
import Rating from '@material-ui/lab/Rating';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import axios from 'axios';
import { getUserProfile } from "../../actions/users";
import { connect } from "react-redux";


const SideBar = styled.div`
  display: flex;
    > div:first-child {
    margin-right: 30px;
    max-width: 350px;
    }
`

const RightSide = styled.div`
  width: 3000px;
`
const FlexDiv = styled.div`
  display: flex;
  justify-content: space-between;
`

const UnstyledLink = styled(Link)`
    color: inherit;
    text-decoration: inherit;
`

export class ProfileCard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      _editing: false,
      tempAge: '',
      tempCollege: '',
      tempBio: '',
    }
  }
  
  toggle_Editing = async() => {
    this.setState(prevState => ({
      _editing: !prevState._editing
    }));
  }

	onChange = async (e) => {
    this.setState({[e.target.id]:e.target.value});
  }

/*
const ProfileCard = props => {
    const { name, age, college , products, bio, wishlist, rating, email, transactions } = props;

  };
*/

  onSubmit = async (e) => {
    e.preventDefault();
    const updateInfo = {
			age : this.state.tempAge,
		  college :this.state.tempCollege,
      bio : this.state.tempBio,
  }
   // this.setState(updateInfo)
    this.toggle_Editing()
    // save the updateInfo to the database
    axios.post(`/api/users/${this.props.id}/edit`, updateInfo)
  .then(() => 
    this.props.getUserProfile(this.props.id))
  .catch(console.log)

  };

  render() { 
    console.log(this.props)
    const {age, college, bio , transactions, email, products, wishlist, rating, name} = this.props

    const products_li = products.map((items, index) => {
      return <li key={items._id + index}>
        {items.name}
        </li>
    });
    
    const wishlist_li = wishlist.map((element, index) => {
      return <li key={element._id + index}>
        {element.item}
      </li>
    });

    const transactions_li = transactions.map(transaction => {
      return  (
        <FlexDiv>
          <UnstyledLink to={`/products/${transaction.item}`}>
            <ListItem>
              <ListItemAvatar>
                <Avatar>
                </Avatar>
              </ListItemAvatar>
              <ListItemText
                primary={transaction.item}
              />
            </ListItem>
          </UnstyledLink>
          <FlexDiv>
            <CheckIcon />
            <CancelIcon />
          </FlexDiv>
        </FlexDiv>
      );
    });

    return (
      <div style={{display: 'flex', flexDirection: 'row'}}>
      <SideBar>
      <Card>
        <CardMedia>
          <img src="https://icon-library.net/images/default-profile-icon/default-profile-icon-24.jpg"  width="350" height="350" alt="profile_image" />
        </CardMedia>
        <CardContent>
          <Typography variant="body2" color="textSecondary" component="p">
            {email} <br/>
            {college} <br/>
            {age} years old<br/>
            {bio}<br></br>
          </Typography>
          <Button align="center" variant="outlined" color="primary" focusVisible onClick={(this.toggle_Editing)}>
                      Edit Profile 
          </Button>  
          {this.state._editing === true ? 
          <div>
            <Dialog open={this.state._editing} onClose={this.toggle_Editing} aria-labelledby="form-dialog-title">
            <DialogTitle id="form-dialog-title">Edit Profile</DialogTitle>
            <DialogContent>
              <DialogContentText>
                Edit your profile and press submit to save your changes
              </DialogContentText>
                <form  onSubmit={this.onSubmit}>
                  <TextField
                    onChange={this.onChange}
                    value={this.state.tempAge || age}
                    id="tempAge"
                    label="Age"
                    type="number"
                    margin="normal"
                    fullWidth
                  />
					        <br/>
                  <TextField
                    onChange={this.onChange}
                    value={this.state.tempCollege || college}
                    id="tempCollege"
                    label="College"
                    type="text"
                    margin="normal"
                    fullWidth
                  />
					        <br/>
                  <TextField
                    onChange={this.onChange}
                    value={this.state.tempBio || bio}
                    id="tempBio"
                    type="text"
                    label="Bio"
                    margin="normal"
                    fullWidth
                  />
					        <br/>
                    <div>
                    <DialogActions>
                  <Button type="reset" onClick={this.toggle_Editing} color="primary">
                    Cancel
                  </Button>
                  <Button type="submit"  color="primary">
                    Submit
                  </Button>
                </DialogActions>
                    </div>
              </form>
          </DialogContent>
          </Dialog>
          </div> 
          : null }
        
        </CardContent>
      </Card>
    </SideBar>
    <RightSide>
      <Card>
        <CardContent>
          <Typography gutterBottom variant="h2" component="h2">
            {name} 
            <br/>
          </Typography>
          <Typography gutterBottom variant="h4" component="h2">
            Rating : {rating} 
            <br/>
            <Box component="span" m={1} borderColor="transparent">
              <Rating name="read-only" value={rating} readOnly />
            </Box>
          </Typography>
          <Typography gutterBottom variant="h4" component="h2">
            Lending List 
            <br/> 
          </Typography>
          <Typography variant="body2" color="textSecondary" component="p" id='products-list'>
            {products_li}       
            <br/>
          </Typography>
            <Button variant="outlined" color="primary">
              Edit Lending List
            </Button> 
            <br/>
            <br/>
          <Typography gutterBottom variant="h4" component="h2">
            Wishlist
            <br/>
          </Typography>
          <Typography variant="body2" color="textSecondary" component="p" id='wish-list'>
            {wishlist_li}
            <br/>
            <Button variant="outlined" color="primary">
              Edit Wishist
            </Button> 
          </Typography>
          <Typography gutterBottom variant="h4" component="h2">
            Transactions
            <br/>
            
          </Typography>
          <div>
            {transactions_li}
          </div>
          </CardContent>
        </Card>
      </RightSide>
    </div>
    );
  };
}


export default connect(undefined,{getUserProfile})(ProfileCard);