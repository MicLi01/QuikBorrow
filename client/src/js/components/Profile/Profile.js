import React, { Component } from "react";
import ProfileCard from './ProfileCard'
import { connect } from "react-redux";
import { fetchProducts } from "../../actions";
import { getUserProfile } from "../../actions/users";

export class UserProfile extends Component {
    constructor(props) {
        super(props);
        this.state = {
            age: 21,
            college:'City College of New York',
            bio: 'Currently a student looking for a full time job',
            wishlist: [{
                _id:1,
                item:"pokemon cards"
                },{
                _id:2,
                item:"calculator"
                }] 
        }
    }

    componentDidMount(){
        this.props.getUserProfile(this.props.match.params.profileId) // getting userProfile based on Id
        this.props.fetchProducts()
    }

    render() {
        let {age , college,bio, wishlist } = this.state
        let {_id, name, rating, email} = this.props.user 
        let mylist = this.props.products
        
        const lendinglist = []

        for (let i =0; i < mylist.length; ++i){
            if(_id === mylist[i].user){
                lendinglist.push(mylist[i])
            }
        }
        
        if (!_id) {
            return <h3> User profile does not exist</h3>
        }
        else {
            return (
            <ProfileCard
            name={name}
            age={age}
            college={college}
            products={lendinglist}
            bio={bio}
            wishlist={wishlist}
            rating={rating}
            email={email}
            />
                )
        }
    }
}


function mapStateToProps(state) {
    return { user : state.user ,
             products : state.products }
}

export default connect(mapStateToProps, { getUserProfile, fetchProducts })(UserProfile);