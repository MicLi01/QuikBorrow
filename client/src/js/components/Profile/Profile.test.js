import React from 'react';
import renderer from 'react-test-renderer';
import Profile from './Profile';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

const createStore = configureMockStore()


jest.mock('../../actions/products', () => ({
    getUserProfile: () => ({type:1}), fetchProducts: () => ({type:1}), fetchTransactions: () => ({type:1})
}));
jest.mock('../../actions/users', () => ({
    getUserProfile: () => ({type:1}), fetchProducts: () => ({type:1}), fetchTransactions: () => ({type:1})
}));


describe('Profile', () => {
    const match = { params: { profileId: 'foo' } }
    
    
    test('render when user match', () => {
        const store = createStore({
            user: {
                _id: '111',
            },
            products: [],
            auth: {user: {id: '001'} },
            transactions:{data: [] }
        })
        let component = renderer.create(
            <Provider store={store}>
            <Profile 
            match = {match}
            />
        </Provider>);
        let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
    });

    test('render when user does not exist', () => {
        const store = createStore({
            user: 0,
            products: [],
            auth: {user: {id: '001'} },
            transactions:{data: [] }
        })
        let component = renderer.create(
            <Provider store={store}>
            <Profile 
            match = {match}
            />
        </Provider>);
       let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('render mylist', () => {
        const store = createStore({
            user: {
                _id: '111',
            },
            products:[
                {user:'213421', item:'chair'},
                {user:'341234', item:'table'}
            ],
            auth: {user: {id: '001'} },
            transactions:{data: [] }
        })
        let component = renderer.create(
            <Provider store={store}>
            <Profile 
            match = {match}
            />
        </Provider>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
});
