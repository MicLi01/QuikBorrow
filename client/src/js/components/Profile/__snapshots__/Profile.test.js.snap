// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile render mylist 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <div
    className="sc-bdVaJa kHPzNS"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation1 MuiCard-root MuiPaper-rounded"
    >
      <div
        className="MuiCardMedia-root"
      >
        <img
          alt="profile_image"
          height="350"
          src="https://icon-library.net/images/default-profile-icon/default-profile-icon-24.jpg"
          width="350"
        />
      </div>
      <div
        className="MuiCardContent-root"
      >
        <p
          className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
        >
           
          <br />
           
          <br />
           years old
          <br />
          <br />
        </p>
        <button
          align="center"
          className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
          disabled={false}
          focusVisible={true}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Edit Profile
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    className="sc-bwzfXH iJYcqs"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation1 MuiCard-root MuiPaper-rounded"
    >
      <div
        className="MuiCardContent-root"
      >
        <h2
          className="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom"
        >
          <br />
        </h2>
        <h2
          className="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom"
        >
          Rating : 
          <br />
          <span
            className="MuiBox-root MuiBox-root-131"
          >
            <span
              aria-label="0 Stars"
              className="MuiRating-root MuiRating-readOnly"
              onMouseLeave={[Function]}
              onMouseMove={[Function]}
              role="img"
            >
              <span
                className="MuiRating-icon MuiRating-iconEmpty"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                className="MuiRating-icon MuiRating-iconEmpty"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                className="MuiRating-icon MuiRating-iconEmpty"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                className="MuiRating-icon MuiRating-iconEmpty"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                className="MuiRating-icon MuiRating-iconEmpty"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
          </span>
        </h2>
        <h2
          className="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom"
        >
          Lending List
          <br />
        </h2>
        <p
          className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
          id="products-list"
        >
          <br />
        </p>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Edit Lending List
          </span>
        </button>
        <br />
        <br />
        <h2
          className="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom"
        >
          Wishlist
          <br />
        </h2>
        <p
          className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
          id="wish-list"
        >
          <li>
            pokemon cards
          </li>
          <li>
            calculator
          </li>
          <br />
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Edit Wishist
            </span>
          </button>
        </p>
        <h2
          className="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom"
        >
          Transactions
          <br />
        </h2>
        <div />
      </div>
    </div>
  </div>
</div>
`;

exports[`Profile render when user does not exist 1`] = `
<h3>
   User profile does not exist
</h3>
`;

exports[`Profile render when user match 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <div
    className="sc-bdVaJa kHPzNS"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation1 MuiCard-root MuiPaper-rounded"
    >
      <div
        className="MuiCardMedia-root"
      >
        <img
          alt="profile_image"
          height="350"
          src="https://icon-library.net/images/default-profile-icon/default-profile-icon-24.jpg"
          width="350"
        />
      </div>
      <div
        className="MuiCardContent-root"
      >
        <p
          className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
        >
           
          <br />
           
          <br />
           years old
          <br />
          <br />
        </p>
        <button
          align="center"
          className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
          disabled={false}
          focusVisible={true}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Edit Profile
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    className="sc-bwzfXH iJYcqs"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation1 MuiCard-root MuiPaper-rounded"
    >
      <div
        className="MuiCardContent-root"
      >
        <h2
          className="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom"
        >
          <br />
        </h2>
        <h2
          className="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom"
        >
          Rating : 
          <br />
          <span
            className="MuiBox-root MuiBox-root-95"
          >
            <span
              aria-label="0 Stars"
              className="MuiRating-root MuiRating-readOnly"
              onMouseLeave={[Function]}
              onMouseMove={[Function]}
              role="img"
            >
              <span
                className="MuiRating-icon MuiRating-iconEmpty"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                className="MuiRating-icon MuiRating-iconEmpty"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                className="MuiRating-icon MuiRating-iconEmpty"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                className="MuiRating-icon MuiRating-iconEmpty"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                className="MuiRating-icon MuiRating-iconEmpty"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
          </span>
        </h2>
        <h2
          className="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom"
        >
          Lending List
          <br />
        </h2>
        <p
          className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
          id="products-list"
        >
          <br />
        </p>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Edit Lending List
          </span>
        </button>
        <br />
        <br />
        <h2
          className="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom"
        >
          Wishlist
          <br />
        </h2>
        <p
          className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
          id="wish-list"
        >
          <li>
            pokemon cards
          </li>
          <li>
            calculator
          </li>
          <br />
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Edit Wishist
            </span>
          </button>
        </p>
        <h2
          className="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom"
        >
          Transactions
          <br />
        </h2>
        <div />
      </div>
    </div>
  </div>
</div>
`;
